rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,562 @@
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix)
+
+#### Current tools
+
+We collected some Tools that work directly with the nix AST and comments.
+Those would benefit from a standardized doc-comment.
+
+- [noogle](https://noogle.dev) - Nix API search engine. It allows searching functions and other expressions.
+- [nix-doc](https://github.com/lf-/nix-doc) - A Nix developer tool leveraging the rnix Nix parser for intelligent documentation search and tags generation
+- [manix](https://github.com/mlvzk/manix) - A fast CLI documentation searcher for Nix.
+
+# Detailed design
+[design]: #detailed-design
+
+Each subsection here contains a decision along with arguments and counter-arguments for (+) and against (-) that decision.
+
+> Note: This RFC does not promote any tool; It focuses only on the format, not implementation. At the same time, this RFC must be technically feasible.
+
+## Discontinue nixdoc's custom format
+
+**Observing**: Nixdoc currently provides some format that can generate documentation from a small fraction of files in nixpkgs, Namely the files for `.lib` functions. The directory provides a consistent community convention that allows the building of static documentation. Applying nixdoc to files outside that directory is only partially possible due to the lack of a standard doc-comment format.
+
+**Considering**: Should we propose the `nixdoc` format as a community standard?
+
+**Decision**: Discontinue nixdoc's custom format in favor of using something more generic.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) Stays primarily compatible with the currently used comments in lib.
+    - (-) Outside the ./lib directory, nixodc is not a convention.
+- (-) Is not widely understood.
+- (-) The format is specific, should be more generic, and follow established conventions. (e.g., Markdown, Doxygen, JSdoc, Haddock)
+- (-) The format is too complex and needs a whole list of complex rules to describe the exact format and content rules; this can also be observed in many false-formatted comments that do not render correctly.
+    - (-) The sections are implicitly continued.
+    - (-) The predefined section headers/keywords are very close to natural language and may accidentally be used.
+        - (-) Syntax to start new sections does not follow conventions (e.g., `Example:`) In contrast JSdoc defines `@Example`; Markdown `# Example`; Doxygen `@example`; Haskell-Haddock `>>>`
+    - (-) The content is implicitly rendered differently depending on predefined rules. Those rules need to be understood/feel natural.
+- (+) Some usages outside of /lib directory indicate the process of adoption.
+- (-) Some multiline comments become doc-comments, but it is unclear which comments have such a property.
+    
+</details>
+
+## `/**` to start a doc-comment
```

Comment:
As discussed OOB: I'd shorten this and the following sections, and move them to somewhere else, where we gather the currently agreed-upon features of doc-comments, namely:

- inner format is CommonMark
- outer format is `/**`
- doc-comments go before assignment expression, not where lambdas are defined

Reply:
We still need to discuss the last point in the next shepherd meeting: static and dynamic implementations should have POCs where we're sure they work and can be extended in the future.